### PR TITLE
Movie Refactor

### PR DESCRIPTION
**Pull request by michaeljevans**
Heavy refactoring of MoviesController
-Create Movie service
-Create Movie and Movies facades
-Refactor Movie, Cast, and Review POROs
-Make structure/formatting changes to movie show page

### CODE REVIEW

File: app/controllers/movies_controller.rb [hunk 1]
```diff
@@ -1,81 +1,14 @@
...
-    response_review = connection.get("/3/movie/#{id}/reviews") do |f|
-      f.headers['Content-Type'] = 'application/json'
-      f.params['api_key']       = ENV['TMDB_API_KEY']
-    end
-    parsed_review = JSON.parse(response_review.body, symbolize_names: true)
-    @review = Review.new(parsed_review[:results])
+  def show
+    @movie = MovieFacade.new(params[:id])
   end
 end
```
Comment (michaeljevans) [inline app/controllers/movies_controller.rb:14]: so much cleaner!

File: app/controllers/movies_controller.rb [hunk 2]
```diff
@@ -1,81 +1,14 @@
...
-
-    response_review = connection.get("/3/movie/#{id}/reviews") do |f|
-      f.headers['Content-Type'] = 'application/json'
-      f.params['api_key']       = ENV['TMDB_API_KEY']
-    end
-    parsed_review = JSON.parse(response_review.body, symbolize_names: true)
-    @review = Review.new(parsed_review[:results])
+  def show
+    @movie = MovieFacade.new(params[:id])
   end
```
Comment (janegreene) [inline app/controllers/movies_controller.rb:13]: This looks so fit now.

File: app/models/friendship.rb [hunk 3]
```diff
@@ -3,8 +3,8 @@ class Friendship < ApplicationRecord
   belongs_to :friend, class_name: 'User'
 
   def self.create_mutual_friendship(current_user_id, friend_id)
-    friendships = [{user_id: current_user_id, friend_id: friend_id},
-                   {user_id: friend_id, friend_id: current_user_id}]
+    friendships = [{ user_id: current_user_id, friend_id: friend_id },
+                   { user_id: friend_id, friend_id: current_user_id }]
```
Comment (michaeljevans) [inline app/models/friendship.rb:7]: you're welcome, Rubocop

File: app/poros/cast.rb [hunk 4]
```diff
@@ -1,9 +1,13 @@
...
   def initialize(cast_info)
-    @members = cast_info.first(10).inject({}) do |acc, cast_member|
-      acc[cast_member[:name]] =  cast_member[:character]
-      acc
+    @members = cast_members(cast_info)
+  end
+
+  def cast_members(cast_info)
+    cast_info.first(10).each_with_object({}) do |member, obj|
+      obj[member[:name]] = member[:character]
```
Comment (michaeljevans) [inline app/poros/cast.rb:10]: Rubocop got mad at .inject - turns out .each_with_object is easier to use, too

File: app/facades/movies_facade.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
+class MoviesFacade
+  attr_reader :movie_service
```
Comment (janegreene) [inline app/facades/movies_facade.rb:2]: Returns search and top 40, when more than one movie is being displayed. Does not include the level of detail necessary for movie detail page, thus the two facades.

File: app/poros/review.rb [hunk 6]
```diff
@@ -1,9 +1,8 @@
...
+
   def initialize(review_info)
-    @reviews = review_info.inject({}) do |acc, review|
-      acc[review[:author]] = review[:content]
-      acc
-    end
+    @author = review_info[:author]
+    @content = review_info[:content]
   end
 end
```
Comment (michaeljevans) [inline app/poros/review.rb:8]: refactored to coincide with a singular review - movie facade now returns an array of review objects

File: app/poros/cast.rb [hunk 7]
```diff
@@ -1,9 +1,13 @@
...
+
   def initialize(cast_info)
-    @members = cast_info.first(10).inject({}) do |acc, cast_member|
-      acc[cast_member[:name]] =  cast_member[:character]
-      acc
+    @members = cast_members(cast_info)
+  end
+
+  def cast_members(cast_info)
+    cast_info.first(10).each_with_object({}) do |member, obj|
```
Comment (janegreene) [inline app/poros/cast.rb:9]: The empty hash is the obj. 
Rubocop did not like the enumerable 'inject' here.

File: app/services/movie_service.rb [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+  def movie_search(terms, page)
+    connection.get('/3/search/movie') do |f|
+      f.headers['Content-Type'] = 'application/json'
+      f.params['api_key']       = ENV['TMDB_API_KEY']
+      f.params['query']         = terms
+      f.params['page']          = page
+    end
+  end
+end
```
Comment (michaeljevans) [inline app/services/movie_service.rb:75]: there is probably more refactoring to be done here - would love to hear suggestions

File: app/poros/review.rb [hunk 9]
```diff
@@ -1,9 +1,8 @@
 class Review
-  attr_reader :reviews
+  attr_reader :author, :content
+
   def initialize(review_info)
```
Comment (janegreene) [inline app/poros/review.rb:4]: Need to add sad path for no reviews testing.

File: app/facades/movie_facade.rb [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+  def cast_details(movie_id)
+    cast_info = @movie_service.cast_details(movie_id)
+    Cast.new(cast_info)
+  end
+
+  def movie_reviews(movie_id)
+    reviews_info = @movie_service.reviews_details(movie_id)
+    reviews_info.map { |review_info| Review.new(review_info) }
+  end
+end
```
Comment (michaeljevans) [inline app/facades/movie_facade.rb:25]: this facade allows us to send one super object to the view!

File: app/services/movie_service.rb [hunk 11]
```diff
@@ -0,0 +1,75 @@
+class MovieService
+  def top_40
+    page1 = top_rated('1')
```
Comment (janegreene) [inline app/services/movie_service.rb:3]: Nice space saving measures!

File: app/views/movies/show.html.erb [hunk 12]
```diff
@@ -1,34 +1,48 @@
...
+  <% end %>
 </section>
-<section class = 'reviews'>
-  <% @review.reviews.each do |author, content| %>
-      <p>Author: <%= author %>
-      <%= content %>
-      </p>
-    <% end %>
+
+<section class='reviews'>
```
Comment (michaeljevans) [inline app/views/movies/show.html.erb:37]: need to add logic here for edge case where there are no reviews for the movie

**Pull request merged**